fast64_internal/sm64/sm64_anim.py

```diff
@@ -950,7 +951,43 @@ def execute(self, context):
             if type(armatureObj.data) is not bpy.types.Armature:
                 raise PluginError("Armature not selected.")
 
-            importAnimationToBlender(romfileSrc, animStart, armatureObj, segmentData, context.scene.isDMAImport)
+            importAnimationToBlender(romfileSrc, animStart, armatureObj, segmentData, context.scene.isDMAImport, "sm64_anim")
+            romfileSrc.close()
+            self.report({"INFO"}, "Success!")
+        except Exception as e:
+            if romfileSrc is not None:
+                romfileSrc.close()
+            raisePluginError(self, e)
+            return {"CANCELLED"}  # must return a set
+
+        return {"FINISHED"}  # must return a set
+
+
+class SM64_ImportAllMarioAnims(bpy.types.Operator):
+    bl_idname = "object.sm64_import_mario_anims"
+    bl_label = "Import All Mario Animations"
+    bl_options = {"REGISTER", "UNDO", "PRESET"}
+
+    # Called on demand (i.e. button press, menu item)
+    # Can also be called from operator search menu (Spacebar)
+    def execute(self, context):
+        romfileSrc = None
+        try:
+            checkExpanded(bpy.path.abspath(context.scene.importRom))
+            romfileSrc = open(bpy.path.abspath(context.scene.importRom), "rb")
+        except Exception as e:
+            raisePluginError(self, e)
+            return {"CANCELLED"}
+        try:
+            if len(context.selected_objects) == 0:
+                raise PluginError("Armature not selected.")
+            armatureObj = context.active_object
+            if type(armatureObj.data) is not bpy.types.Armature:
+                raise PluginError("Armature not selected.")
+
+            for animName, adress in marioAnimations:
+                importAnimationToBlender(romfileSrc, adress, armatureObj, segmentData, context.scene.isDMAImport, animName)
```

Comment:
what is `segmentData` here?

Reply:
segmentData is an object describing the segment locations in sm64. It has type dict[segNum] = (segStart, segEnd) and all of them are ints

Reply:
it seems like segment data is inherited from the global context since it is not defined in that function, which should not affect any operation as far as mario animation importing goes.
other anim importing functions run this to get the segmentData
```
            levelParsed = parseLevelAtPointer(romfileSrc, level_pointers[context.scene.levelAnimImport])
            segmentData = levelParsed.segmentData
```

Reply:
thanks
it's not some kind of fast64 global, is it?
I don't see it set in this function

Reply:
I'm not entirely certain as I didn't trace the code, but I do know that mario animations are in the ROM and not in any specific segment, so the actual data shouldn't be needed. I would change it to an empty dict if possible.